sandbox/KalturaMediaPlugin/src/com/wowza/kaltura/KalturaMediaModule.java

```diff
@@ -0,0 +1,284 @@
+import java.net.HttpURLConnection;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.Hashtable;
+
+import com.wowza.wms.amf.AMFDataList;
+import com.wowza.wms.amf.AMFDataObj;
+import com.wowza.wms.amf.AMFPacket;
+import com.wowza.wms.application.IApplicationInstance;
+import com.wowza.wms.application.WMSProperties;
+import com.wowza.wms.client.IClient;
+import com.wowza.wms.logging.WMSLoggerFactory;
+import com.wowza.wms.media.model.MediaCodecInfoAudio;
+import com.wowza.wms.media.model.MediaCodecInfoVideo;
+import com.wowza.wms.module.ModuleBase;
+import com.wowza.wms.request.RequestFunction;
+import com.wowza.wms.rtp.model.RTPSession;
+import com.wowza.wms.stream.IMediaStream;
+import com.wowza.wms.stream.IMediaStreamActionNotify;
+import com.wowza.wms.stream.IMediaStreamActionNotify3;
+
+public class KalturaMediaModule extends ModuleBase  implements IMediaStreamActionNotify3
+{
+	static final String methodPublish = "publish";
+
+	static final String methodUnpublish = "unpublish";
+	
+	static final String methodAuthenticate = "authenticate";
+
+	private static class KalturaMediaServerConfig {
+
+		public KalturaMediaServerConfig(String path)
+		{
+			try
+			{
+				BufferedReader in = new BufferedReader( new FileReader(path) );
+				String inpitLine;
+				while((inpitLine = in.readLine()) != null){
+					String [] args = inpitLine.split("=");
+					switch(args[0])
+					{
+					case "SERVER_NODE_NAME":
+						hostName = nodeName = args[1];
+						break;
+					case "apphome_url":
+						kalturaAPIServerURL = args[1];
+						break;
+					}
+				}
+				in.close();
+				hostName = InetAddress.getLocalHost().getHostName();
+				Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+				while (interfaces.hasMoreElements()){
+				    NetworkInterface current = interfaces.nextElement();
+				    System.out.println(current);
+				    if (!current.isUp() || current.isLoopback() || current.isVirtual()) continue;
+				    Enumeration<InetAddress> addresses = current.getInetAddresses();
+				    while (addresses.hasMoreElements()){
+				        InetAddress current_addr = addresses.nextElement();
+				        if(!(current_addr instanceof Inet4Address))
+				        	continue;
+			//	        if(current_addr.isSiteLocalAddress())
+				//        	continue;
+				        if (current_addr.isLoopbackAddress()) 
+				        	continue;
+				        WMSLoggerFactory.getLogger(null).warn("KalturaMediaServerConfig : hostName: " + current_addr.getHostAddress());
+				        System.out.println(current_addr.getCanonicalHostName());
+				        hostName = current_addr.getHostAddress();
+				        break;
+				    }
+				}
+				
+			}
+			catch(Exception e)
+			{
+
+			}
+		}
+
+		private static final KalturaMediaServerConfig s_Instance = new KalturaMediaServerConfig("/opt/kaltura/app/configurations/ecdn.ini");
```

Comment:
It doesn't make sense to use ecdn.ini in SaaS.
If needed, create a new ini file to be used both in SaaS and eCDN.